### PR TITLE
add LDAP and Slack Support

### PR DESCRIPTION
**Pull request by LaysDragon**
add Slack plugin support

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -26,7 +26,20 @@ ENV \
   RABBIT_HOST="rabbit:5672" \
   REDIS_HOST="redis:6379" \
   SAML_AUTH_ENABLE="False" \
-  SAML_AUTH_JSON_CONFIG=""
+  SAML_AUTH_JSON_CONFIG=""\
```
Comment (mvitale1989) [inline Dockerfile]: A default for `SLACK_INTEGRATION_ENABLE` would be good to have here.
```suggestion
  SAML_AUTH_JSON_CONFIG="" \
  SLACK_INTEGRATION_ENABLE="False" \
```

File: conf/taiga/docker-settings.py [hunk 2]
```diff
@@ -107,3 +107,40 @@ def load_file(path):
...
+  # These fields need to have a value in LDAP 
+  LDAP_USERNAME_ATTRIBUTE = os.getenv("LDAP_USERNAME_ATTRIBUTE")
+  LDAP_EMAIL_ATTRIBUTE = os.getenv("LDAP_EMAIL_ATTRIBUTE")
+  LDAP_FULL_NAME_ATTRIBUTE = os.getenv("LDAP_FULL_NAME_ATTRIBUTE")
+
+#########################################
+## SLACK Integration
+#########################################
+
+INSTALLED_APPS += ["taiga_contrib_slack"]
```
Comment (mvitale1989) [inline conf/taiga/docker-settings.py]: This is not truly disabled unless conditionally left out.
```suggestion
if os.getenv("SLACK_INTEGRATION_ENABLE").lower() == "true":
  INSTALLED_APPS += ["taiga_contrib_slack"]
```

File: conf/taiga/docker-settings.py [hunk 3]
```diff
@@ -107,3 +107,40 @@ def load_file(path):
...
+
+
+if os.getenv("LDAP_AUTH_ENABLE").lower() == "true":
+  INSTALLED_APPS += ["taiga_contrib_ldap_auth_ext"]
+  # TODO https://github.com/Monogramm/taiga-contrib-ldap-auth-ext/issues/16
+  LDAP_SERVER = os.getenv("LDAP_SERVER")
+  LDAP_PORT = int(os.getenv("LDAP_PORT"))
+
+  # Flag to enable LDAP with STARTTLS before bind
+  LDAP_START_TLS = os.getenv("LDAP_START_TLS")
```
Comment (mvitale1989) [inline conf/taiga/docker-settings.py]: Underlying module expects a true boolean for this, so this conversion is needed
```suggestion
  LDAP_START_TLS = True if os.getenv("LDAP_START_TLS").lower() == "true" else False
```

File: README.md [hunk 4]
```diff
@@ -143,6 +143,18 @@ at [this example](https://github.com/benhutchins/docker-taiga-example/tree/maste
...
+## Configuring LDAP
+If you want to use LDAP authenticate, you'll need to specify all of
+these additional arguments to your `docker run` command.
+
+- `-e LDAP_AUTH_ENABLE=True`
+- `-e LDAP_BIND_DN=<bind account dn>`
+- `-e LDAP_SEARCH_BASE=<base>` 
+- `-e LDAP_SERVER=ldap://<ldap_server_host_name>`
+
+Other LDAP option can take a look at plugin docs <https://github.com/ensky/taiga-contrib-ldap-auth>
```
Comment (mvitale1989) [inline README.md]: This link references a different ldap plugin than the one that's being used
```suggestion
Other LDAP option can take a look at plugin docs <https://github.com/Monogramm/taiga-contrib-ldap-auth-ext>
```

File: README.md [hunk 5]
```diff
@@ -143,6 +143,18 @@ at [this example](https://github.com/benhutchins/docker-taiga-example/tree/maste
 setup where it specifies a `taiga-conf/local.py`. You can then configure this
 and other [settings available in Taiga](https://github.com/taigaio/taiga-back/blob/master/settings/local.py.example).
 
+## Configuring LDAP
+If you want to use LDAP authenticate, you'll need to specify all of
+these additional arguments to your `docker run` command.
+
+- `-e LDAP_AUTH_ENABLE=True`
+- `-e LDAP_BIND_DN=<bind account dn>`
```
Comment (mvitale1989) [inline README.md:151]: Would add `LDAP_BIND_PASSWORD` here as well, to make it easier to remember for users.
```suggestion
- `-e LDAP_BIND_DN=<bind account dn>`
- `-e LDAP_BIND_PASSWORD=<bind password>`
```

**Review by mvitale1989 — CHANGES_REQUESTED** [review]
Looks good, although after some tests i found a few spots that need fixing to make things fully work. After those are done i'd be ready for merging

File: README.md (hunk 4 above)
Comment (LaysDragon) [reply README.md]: oh,you are right ,I pasted the wrong link

File: conf/taiga/docker-settings.py (hunk 2 above)
Comment (LaysDragon) [reply conf/taiga/docker-settings.py]: oh yes, I forget to add the condition

File: Dockerfile (hunk 1 above)
Comment (LaysDragon) [reply Dockerfile]: ok

File: conf/taiga/docker-settings.py (hunk 3 above)
Comment (LaysDragon) [reply conf/taiga/docker-settings.py]: nice detail investigation

File: README.md (hunk 5 above)
Comment (LaysDragon) [reply README.md:151]: ok that improved the document

**Comment by LaysDragon** [comment]
OK I have already accept those changes :D

**Review by mvitale1989 — APPROVED** [review]

**Pull request merged**